src/cmdline.rs

```diff
@@ -329,6 +348,82 @@ pub fn parse_args() -> Result<Config> {
                         ),
                 ),
         )
+        .subcommand(
+            SubCommand::with_name("osmet")
+                .about("Efficient CoreOS metal disk image packing using OSTree commits")
+                // users shouldn't be interacting with this command normally
+                .setting(AppSettings::Hidden)
+                .subcommand(
+                    SubCommand::with_name("pack")
+                        .about("Create osmet file from CoreOS block device")
+                        .arg(
+                            Arg::with_name("output")
+                                .long("output")
+                                .value_name("FILE")
+                                .required(true) // could output to stdout if missing?
+                                .help("Path to osmet file to write")
+                                .takes_value(true),
+                        )
+                        .arg(
+                            // XXX: rebase on top of
+                            // https://github.com/coreos/coreos-installer/pull/178 and use the same
```

Comment:
That PR takes SHA512 only (like Ignition), though. Should we switch this PR to align on that?

Reply:
I mostly meant using the same `type-digest` format and possibly factoring out the `IgnitionHash` bit and its `try_parse` there to handle SHA256. In practice, cosa stores SHA256 digests for artifacts so that's likely what'll get passed in so we don't have to recalculate a different digest. And OSTree filepath objects are also SHA256. But at least encoding `sha256` in the argument string will make it that bit easier to move to `sha512` or whatever else one day.

Reply:
#178 is merged, though let's address this in a follow-up to try to get this in sooner. Going to mark this as resolved for now to help track progress here.